tests/test_resolver.py

```diff
@@ -299,6 +300,43 @@ def test_combine_install_requirements(repository, from_line):
     assert str(combined_all.req.specifier) == "<3.2,==3.1.1,>3.0"
 
 
+def test_combine_install_requirements_extras(repository, from_line, make_package):
```

Comment:
Could you split each test block into different tests or parametrize tests if possible?

Reply:
Sounds good. I split it into two tests.